---
title: "SORCS2_Genetics"
author: "SteveYounkin"
date: "October 21, 2015"
output: html_document
---

##Analysis Plan

We will pursue a simple, straightforward analysis of our SORCS2 genotypes using a robust, R-based reproducible research approach. This will (i) provide a single document that gives us a complete account of our genetic analysis of SORCS2 and (ii) familiarize us with a scalable, cutting-edge  approach that is capable of handling far more complicated analyses on large datasets.

We will keep all of our work in one R Markdown document: SOCS2_Genetics.Rmd.  

To communicate with one another as we build this document, we will use the SORCS2 repository in my github account (https://github.com/StevenGYounkin/SORCS2), and we will all use git (GitHub gui) for version control. 

All files used or generated by SORCS2_Genetics.Rmd will be in this SORCS2 repo. 

Joseph will take responsiblity for providing whatever help may be needed for each of us to interact effectively with one another on GitHub to produce a final document.

Our goal, which we may or may not achieve, will be to complete a "publication ready"" PLoS ONE manuscript by Monday, November 16.

###What each participant needs:
1. RStudio installed 
2. GitHub installed
3. Local clone of the repo at https://github.com/StevenGYounkin/SORCS2
    + SORCS2_Genetics.Rmd, the file we are generating together, is in this repo
        + All files used or generated by SORCS2_Genetics.Rmd are in this repo
4. R packages installed
    + dplyr
    + data.table
5. PLINK installed and set up so it can be called by R with system("plink......")
6. Perl installed and set up so it can be called by R with system("   ")

###Useful documentation:
1. GitHub
    + https://training.github.com/kit/downloads/github-git-cheat-sheet
2. R Markdown
    + https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf
3. dplyr and tidyr
    + https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf
4. PLINK
    + http://pngu.mgh.harvard.edu/~purcell/plink/
    
###Responsibility for Specific Analysis Steps:
1. Read in existing SORCS2 dataset(s) and initate analysis
    + Steve made the Younkin db (Ydb) dataset available, Joseph made the ADSP dataset available
    + Steve developed functions to read in and analyze the SORCS2 datasets from Ydb (ydb2plink) and adsp2plink. 
    + Steve will complete this analysis, determine how best to display manuscript text, write the introduction.
3. Download ADSP SORCS2 dataset
    + Joseph has downloaded dataset for 1000 samples and will do the rest as the data become available (3000 samples may soon be ready)
4. Create Figure 1 showing a SORCS2 gene diagram with information on SNP location etc.
    + Ben and Leonie are working on this with Maarten's help. 
       
###Steve's Content

####Introduction

This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.
        
This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.This is just some text to read in.
        
####Function to read in and analyze initial Ydb SORCS2 dataset analyzed in 10/14 by Chris Medway.

On 10/27/14, I sent an email to Anders to which was attached a flat file with our initial Ydb SORCS2 data and some analyses performed by Chris Medway. This flat file, now named SORCS2_CM.csv, was copied into our SORSC2 repo.

```{r, echo=FALSE}

                #Create function to read in dataset from Chris Medway (10/27/14)
                #make PLINK files (map, fam, cov.txt, and lgen) and analyze them.
cm2plink <- function(input="SORCS2_CM.csv", output="SORCS2_CM") {

                #load requisite packages       
library(data.table)
suppressWarnings(suppressMessages(library(dplyr)))
        
                #Read in dataset generated from the Younkin database
input <- "SORCS2_CM.csv"
data <- read.table(input, sep=',', stringsAsFactors = FALSE, header=T)
#str(data);head(data); tail(data)

                #Create output files
output <- "SORCS2.CM"
out1 <- paste(output,sep="",".map")
out2 <- paste(output,sep="",".fam")
out3 <- paste(output,sep="","_cov.txt")
out4 <- paste(output,sep="",".lgen")
out5 <- paste(output,sep="",".plink.csv")
#print(out1);print(out2);print(out3);print(out4);print(out5)

                #Generate PLINK map file by copying and renaming "SORCS2_YdB.map"
map <- read.table("SORCS2_YdB.map", stringsAsFactors = FALSE, header=F)
#print(map,row.names=F, col.names=F)
write.table(map,out1,sep='\t',quote=FALSE,row.names=F,col.names=F)

        #In his flat file, Chris encoded each SNP as a separate column with 0,1,2 coding
        #There'is additional wrangling to do to produce map, fam, cov, and lgen files

        #Add columns that are not in Chris file and rename as needed
data <- mutate(data, PID=0, MID=0, rs="rs", A1=0, A2=0)
data <- rename (data,IID=SubKey, AFF=Dx2, JS=JS_SIB, NCRAD=INDIANA, Age=DxAge, APOE4dose=APOE4, APOE2dose=APOE2)
#str(data)

        #Recode as needed to conform to PLINK cov and fam conventions
data[data$AFF == 1,][, "AFF"] <- 2
data[data$AFF == 0,][, "AFF"] <- 1

data[data$SEX == 1,][, "SEX"] <- 2
data[data$SEX == 0,][, "SEX"] <- 1

        #Check coding for PLINK
#print(data %>% group_by(AFF,SEX,APOE4dose, APOE2dose,A1,A2) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(A1,A2) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(rs,AFF) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(SEX) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(APOE4dose,APOE2dose) %>% summarize(n=n()))      
#print.data.frame(y)

        #Create PLINK fam file
fam <- select(data,FID,IID,PID,MID,SEX,AFF)
fam <- distinct(fam)
#str(fam); head(fam); tail(fam)
write.table(fam,out2,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2fam=read.table("SORCS2.fam", header = F)
#str(SORCS2fam);head(SORCS2fam);tail(SORCS2fam)

        #Create PLINK cov file; 
        #Not sure why, but errors sometimes occurred in r when using out3.cov" as output using write.table
        #So I used "out3_cov.txt" thereby avoiding this problem
cov <- select(data,FID,IID,JS,RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose)
cov <- distinct(cov)
#str(cov)
write.table(cov,out3,sep='\t',quote=FALSE,row.names=F,col.names=T)
#SORCS2cov=read.table, out3, header = T)
#str(SORCS2cov);head(SORCS2cov); tail(SORCS2cov)

        #Create PLINK lgen file
        #
lgen <- select(data,FID,IID,SORCS2_rs34058821_GA,rs,A1,A2)
#str(lgen)
lgen <- rename(lgen, Call=SORCS2_rs34058821_GA)
#str(lgen)
lgen <- mutate(lgen, rs="rs34058821")
#str(lgen)
call <- c(0,1,2)
lgen <- filter(lgen, Call %in% call)
#print.data.frame(lgen %>% group_by(rs,Call) %>% summarize(n()))

x <- select(data,FID,IID,SORCS2_rs35935435_CA,rs,A1,A2)
#str(x)
x <- rename(x, Call=SORCS2_rs35935435_CA)
#str(x)
x <- mutate(x, rs="rs35935435")
#str(x)
call <- c(0,1,2)
x <- filter(x, Call %in% call)
#print.data.frame(x %>% group_by(rs,Call) %>% summarize(n()))

lgen <- bind_rows(lgen,x)
#print.data.frame(lgen %>% group_by(rs,Call) %>% summarize(n()))

x <- select(data,FID,IID,SORCS2_rs16840892_CT,rs,A1,A2)
#str(x)
x <- rename(x, Call=SORCS2_rs16840892_CT)
#str(x)
x <- mutate(x, rs="rs16840892")
#str(x)
call <- c(0,1,2)
x <- filter(x, Call %in% call)
#print.data.frame(x %>% group_by(rs,Call) %>% summarize(n()))

lgen <- bind_rows(lgen,x)
#print.data.frame(lgen %>% group_by(rs,Call) %>% summarize(n()))

x <- select(data,FID,IID,SORCS2_rs6816604_CT,rs,A1,A2)
#str(x)
x <- rename(x, Call=SORCS2_rs6816604_CT)
#str(x)
x <- mutate(x, rs="rs6816604")
#str(x)
call <- c(0,1,2)
x <- filter(x, Call %in% call)
#print.data.frame(x %>% group_by(rs,Call) %>% summarize(n()))

lgen <- bind_rows(lgen,x)
#print.data.frame(lgen %>% group_by(rs,Call) %>% summarize(n()))

lgen[lgen$Call == 0,][, "A1"] <- 1
lgen[lgen$Call == 1,][, "A1"] <- 1
lgen[lgen$Call == 2,][, "A1"] <- 2

lgen[lgen$Call == 0,][, "A2"] <- 1
lgen[lgen$Call == 1,][, "A2"] <- 2
lgen[lgen$Call == 2,][, "A2"] <- 2

#print.data.frame(lgen %>% group_by(rs,Call,A1,A2) %>% summarize(n()))

lgen <- select(lgen,FID,IID,rs,A1,A2)
#str(lgen)
write.table(lgen,out4,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2lgen=read.table("SORCS2.lgen", header = F)
#str(SORCS2lgen)

assoc <- paste("plink",sep=" ","--lfile", output, "--assoc", "--ci 0.95", "--out", output)
logistic <- paste("plink",sep=" ","--lfile", output, "--logistic", "--covar", out3, "--covar-name RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--out", output)
assocfile <- paste(output,sep="",".assoc")
logisfile <- paste(output,sep="",".assoc.logistic")
system(assoc)
system(logistic)
f1 <- read.table(assocfile, header=T)
#print(f1)
f1 <- f1 %>% select(CHR,SNP,BP,F_A, F_U, OR, P) %>% rename(FREQ_AD=F_A, FREQ_CON=F_U, OR_assoc=OR, P_assoc=P)
print(f1,digits=2,row.names=F)
#head(f1)
f1 <- f1 %>% select(CHR,SNP,BP,FREQ_AD,FREQ_CON)
f2 <- read.table(logisfile, header=T)
#print(f2)
f2 <- f2 %>% filter (TEST=="ADD") %>% select (SNP,NMISS,OR,L95,U95,P) %>% rename(OR_logis=OR, P_logis=P)
#head(f2)
f3<- left_join(f1,f2,by="SNP")
print(f3,digits=2,row.names=F)
write.table(f3,out5,sep=",", quote=FALSE,row.names=F,col.names=T)
#print("PLINK commands:");print(assoc);print(logistic)
}

```
####Analyze SORCS2 dataset from Chris Medway
```{r}
cm2plink(input="SORCS2_CM.csv", output="SORCS2_CM")
```

####Function to read in and analyze Ydb SORCS2 dataset
The data we have on the 4 common SORCS2 missense variants in the Younkin database (YdB) were captured using the following query:

SELECT        TOP (100) PERCENT Younkin.dbo.Subject.SubKey, Younkin.dbo.Subject.SubjectIDType, Younkin.dbo.Subject.SubjectIDValue, Younkin.dbo.Subject.Series, 
                         Younkin.dbo.Subject.Dx, Younkin.dbo.Subject.ApoE, Younkin.dbo.Subject.DxAge, Younkin.dbo.Subject.Gend, Younkin.dbo.Variant.VariantID, 
                         Younkin.dbo.Variant.VariantName, Younkin.dbo.VariantDataAll.Call, Younkin.dbo.VariantDataAll.BestCall, Younkin.dbo.Variant_New.chr, 
                         Younkin.dbo.Variant_New.position, Younkin.dbo.Variant_New.rs
FROM            Younkin.dbo.VariantDataAll INNER JOIN
                         Younkin.dbo.Variant ON Younkin.dbo.VariantDataAll.VariantID = Younkin.dbo.Variant.VariantID INNER JOIN
                         Younkin.dbo.Subject ON Younkin.dbo.VariantDataAll.SKey = Younkin.dbo.Subject.SubKey INNER JOIN
                         Younkin.dbo.Variant_New ON Younkin.dbo.VariantDataAll.VariantID = Younkin.dbo.Variant_New.VariantID
WHERE        (Younkin.dbo.Variant.VariantID = 2069 OR
                         Younkin.dbo.Variant.VariantID = 2070 OR
                         Younkin.dbo.Variant.VariantID = 2072 OR
                         Younkin.dbo.Variant.VariantID = 2073)
ORDER BY Younkin.dbo.Variant.VariantID

The results of this query were copied into SORCS2_Ydb.csv

```{r, echo=FALSE}

                #Create function to read in SQL-derived dataset from Younkin database, 
                #make PLINK files (map, fam, cov.txt, and lgen) and analyze them.
ydb2plink <- function(input="SORCS2_Ydb.csv", output="SORCS2_YdB") {

                #load requisite packages       
library(data.table)
suppressWarnings(suppressMessages(library(dplyr)))
        
                #Read in dataset generated from the Younkin database
data <- read.table(input, sep=',', stringsAsFactors = FALSE, header=T)

                #Create output files
#output <- "SORCS2_Ydb"
out1 <- paste(output,sep="",".map")
out2 <- paste(output,sep="",".fam")
out3 <- paste(output,sep="","_cov.txt")
out4 <- paste(output,sep="",".lgen")
out5 <- paste(output,sep="",".plink.csv")
#print(out1);print(out2);print(out3);print(out4);print(out5)

                #Generate PLINK map file
map <- select(data,chr, rs, position)
map <- mutate(map,morgans=0)
map <- rename(map, bp=position)
map <- select(map,chr,rs,morgans,bp)
map <- distinct(map)
#print(map)
write.table(map,out1,sep='\t',quote=FALSE,row.names=F,col.names=F)

                #Filter, using dplyr package, to include only subjects with: 
                #Dx = AD or CON, Gend = M or F, ApoE = 22,23,24,33,34,or 44 
                #Series = JS,SibPair_White,RS,AUT,Indiana,NW, DxAge >= 65, BestCall = 1
                #Age (at diagnosis for AD/at entry for CON) of 65 or over
dx <- c("AD","CON")
sex <- c("M","F")
apoe <- c(22,23,24,33,34,44)
series <- c("JS", "RS", "AUT", "Indiana", "SibPair_White", "NW")
data <- filter(data, Dx %in% dx, Gend %in% sex, ApoE %in% apoe, Series %in% series, BestCall == 1)
data <-suppressWarnings(mutate(data,Age=as.numeric(DxAge)))
data <- filter (data, Age >= 65)

        #Create columns for PLINK fam, cov, and lgen files
data <- mutate(data, AFF=99, A1=99, A2=99, APOE4dose=99, APOE2dose=99, SEX=99, JS=0, RS=0, AUT=0, NCRAD=0,NW=0, FID=0, IID=SubKey, PID=0, MID=0)

data[data$Dx == "AD",][, "AFF"] <- 2
data[data$Dx == "CON",][, "AFF"] <- 1

data[data$Call == 11,][, "A1"] <- 1
data[data$Call == 12,][, "A1"] <- 1
data[data$Call == 22,][, "A1"] <- 2

data[data$Call == 11,][, "A2"] <- 1
data[data$Call == 12,][, "A2"] <- 2
data[data$Call == 22,][, "A2"] <- 2

data[data$Gend == "M",][, "SEX"] <- 1
data[data$Gend == "F",][, "SEX"] <- 2

data[data$ApoE == 22,][, "APOE4dose"] <- 0
data[data$ApoE == 23,][, "APOE4dose"] <- 0
data[data$ApoE == 24,][, "APOE4dose"] <- 1
data[data$ApoE == 33,][, "APOE4dose"] <- 0
data[data$ApoE == 34,][, "APOE4dose"] <- 1
data[data$ApoE == 44,][, "APOE4dose"] <- 2

data[data$ApoE == 22,][, "APOE2dose"] <- 2
data[data$ApoE == 23,][, "APOE2dose"] <- 1
data[data$ApoE == 24,][, "APOE2dose"] <- 1
data[data$ApoE == 33,][, "APOE2dose"] <- 0
data[data$ApoE == 34,][, "APOE2dose"] <- 0
data[data$ApoE == 44,][, "APOE2dose"] <- 0

data[data$Series == "JS",][, "JS"] <- 1
data[data$Series == "SibPair_White",][, "JS"] <- 1
data[data$Series == "RS",][, "RS"] <- 1
data[data$Series == "AUT",][, "AUT"] <- 1
data[data$Series == "Indiana",][, "NCRAD"] <- 1
data[data$Series == "NW",][, "NW"] <- 1

        #Check coding for PLINK
#print(data %>% group_by(rs,AFF,SEX,APOE4dose, APOE2dose,A1,A2) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(rs,A1,A2) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(rs,AFF) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(rs,SEX) %>% summarize(n=n()))
#print.data.frame(data %>% group_by(APOE4dose,APOE2dose) %>% summarize(n=n()))      
#print.data.frame(y)

        #Create PLINK fam file
fam <- select(data,FID,IID,PID,MID,SEX,AFF)
fam <- distinct(fam)
#str(fam); head(fam); tail(fam)
write.table(fam,out2,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2fam=read.table("SORCS2.fam", header = F)
#str(SORCS2fam);head(SORCS2fam);tail(SORCS2fam)

        #Create PLINK cov file; 
        #Not sure why, but errors sometimes occurred in r when using "SORCS2.cov"" as output using write.table
        #So I used "SORCS2cov.txt" thereby avoiding this problem
cov <- select(data,FID,IID,JS,RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose)
cov <- distinct(cov)
#str(cov)
write.table(cov,out3,sep='\t',quote=FALSE,row.names=F,col.names=T)
#SORCS2cov=read.table("SORCS2cov.txt", header = T)
#str(SORCS2cov);head(SORCS2cov); tail(SORCS2cov)

        #Create PLINK lgen file
lgen <- select(data,FID,IID,rs,A1,A2)
#str(lgen)
write.table(lgen,out4,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2lgen=read.table("SORCS2.lgen", header = F)
#str(SORCS2lgen)

assoc <- paste("plink",sep=" ","--lfile", output, "--assoc", "--ci 0.95", "--out", output)
logistic <- paste("plink",sep=" ","--lfile", output, "--logistic", "--covar", out3, "--covar-name RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--out", output)
assocfile <- paste(output,sep="",".assoc")
logisfile <- paste(output,sep="",".assoc.logistic")
system(assoc)
system(logistic)
f1 <- read.table(assocfile, header=T)
#print(f1)
f1 <- f1 %>% select(CHR,SNP,BP,F_A, F_U, OR, P) %>% rename(FREQ_AD=F_A, FREQ_CON=F_U, OR_assoc=OR, P_assoc=P)
print(f1,digits=2,row.names=F)
#head(f1)
f1 <- f1 %>% select(CHR,SNP,BP,FREQ_AD,FREQ_CON)
f2 <- read.table(logisfile, header=T)
#print(f2)
f2 <- f2 %>% filter (TEST=="ADD") %>% select (SNP,NMISS,OR,L95,U95,P) %>% rename(OR_logis=OR, P_logis=P)
#head(f2)
f3<- left_join(f1,f2,by="SNP")
print(f3,digits=2,row.names=F)
write.table(f3,out5,sep=",", quote=FALSE,row.names=F,col.names=T)
#print("PLINK commands:");print(assoc);print(logistic)
}

```
####Analyze SORCS2 dataset from Ydb
```{r}
ydb2plink(input="SORCS2_Ydb.csv", output="SORCS2_YdB")
#print(assoc)
```
       
        #PLINK --assoc: analyzes standard case/control (allelic) association with LOAD, no covariates
#system("plink --lfile output --assoc --ci 0.95 --out output")
#t1assoc <- read.table(paste(output, sep="", ".assoc"), header = T)
#str(SORCS2assoc);head(SORCS2assoc);tail(SORCS2assoc)

#ydb2plink()

        #PLINK --logistic: analyzes association by logistic regression using an additive model with all covariates
#system("plink --lfile output --logistic --ci 0.95 --covar out3 --covar-name RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose --out output")
#t2logistic <- read.table(paste(output, sep="", ".assoc.logisitc", header = T)

#str(SORCS2logAll);head(SORCS2logAll);tail(SORCS2logAll)
#x <- select (SORCS2logAll, SNP,TEST,NMISS, OR, L95, U95, P)
#print.data.frame(x)
#x <- filter(x, TEST=="ADD")
#print.data.frame(x)

#system("plink --lfile SORCS2 --hap-window 4 --hap-logistic --ci 0.95 --covar SORCS2cov.txt --covar-name RS-APOE2dose --mhf 0.005 --out SORCS2haplogAllCov")
#system("plink --lfile SORCS2 --hap-window 4 --hap-logistic --ci 0.95 --covar SORCS2cov.txt --covar-name RS-APOE2dose --mhf 0.005 --hap-omnibus --out SORCS2haplogAllCovOmni")
#system("plink --lfile SORCS2 --logistic --ci 0.95 --covar SORCS2cov.txt --covar-name APOE4dose,APOE2dose --out SORCS2logAPOECov")
#list.files()
```


####Read in and Analyze ADSP SORCS2 dataset

The data we have on the 4 common SORCS2 missense variants in our ADSP dataset was captured by Joseph Reddy in a flatfile (SORCS2_ff.txt) as described in his section below. I read in the dataset (SORCS2_ff.txt) from the SORSC2 repo.The following function reads the data in, creates PLINK, files and then analyzes them in PLINK as for the SORCS2 dataset from the Younkin database.

```{r}
                #Create function that reads in the ADSP dataset generated by Joseph Reddy and outputs PLINK map, fam, cov.txt, and lgen files
ADSP2plink <- function(file="SORCS2_ff.txt", out1="SORCS2_ADSP.map", out2="SORCS2_ADSP.fam", out3="SORCS2_ADSP_cov.txt", out4="SORCS2_ADSP.lgen") {

                #load requisite packages       
library(data.table)
library(dplyr)        


        #Read in data on 4 common SORCS2 missense variants in the ADSP database        #
ADSP <- read.table("SORCS2_ff.txt", stringsAsFactors = FALSE, header = T)
str(ADSP);head(ADSP);tail(ADSP)

        #Filter, using dplyr package, to include only subjects with: 
        #Case.control = 2 or 1 corresponding to AD or CON, Sex = 0 or 1 (M or F), APOE = 22,23,24,33,34,or 44 
        #Series = ADSP, Age >= 65, Call = 11,12,22
library(dplyr)
dx_1 <- c(2,1,0)
sex_1 <- c(0,1)
apoe_1 <- c(22,23,24,33,34,44)
series_1 <- c("ADSP")
call_1 <- c(11,12,22)
SORCS2_1 <- filter(ADSP, Case.control %in% dx_1, Sex %in% sex_1, APOE %in% apoe_1, Series %in% series_1, Call %in% call_1)
#SORCS2_1 <- mutate(SORCS2,Age=as.numeric(DxAge))
SORCS2_1 <- filter (SORCS2_1, Age >= 65)
str(SORCS2_1);head(SORCS2_1);tail(SORCS2_1)

        #Create PLINK map file
library(data.table)
map <- select(SORCS2_1, SNP_ID, Position)
map <- mutate(map,morgans=0, chr=4)
map <- rename(map, bp=Position, rs=SNP_ID)
map <- select(map,chr,rs,morgans,bp)
map <- distinct(map)
snps <- c("rs6816604","rs35935435","rs16840892","rs34058821")
map <- filter(map,rs %in% snps )
#print(map)
write.table(map,out1,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2map_1=read.table("SORCS2_1.map", header = F)
#print(SORCS2map_1,row.names=F,col.names=F)

        #Create variables for PLINK fam, cov, and lgen files
SORCS2_1 <- mutate(SORCS2_1, AFF=99, A1=99, A2=99, APOE4dose=99, APOE2dose=99, SEX=99, ADSP=0, FID=0, IID=as.character(Subject_ID), PID=0, MID=0)

SORCS2_1[SORCS2_1$Case.control == 0,][, "AFF"] <- 1  #CON=0
SORCS2_1[SORCS2_1$Case.control == 1,][, "AFF"] <- 2  #AD=1
SORCS2_1[SORCS2_1$Case.control == 2,][, "AFF"] <- 2  #probable AD=2 are included as AD for PLINK analysis

SORCS2_1[SORCS2_1$Call == 11,][, "A1"] <- 1
SORCS2_1[SORCS2_1$Call == 12,][, "A1"] <- 1
SORCS2_1[SORCS2_1$Call == 22,][, "A1"] <- 2

SORCS2_1[SORCS2_1$Call == 11,][, "A2"] <- 1
SORCS2_1[SORCS2_1$Call == 12,][, "A2"] <- 2
SORCS2_1[SORCS2_1$Call == 22,][, "A2"] <- 2

SORCS2_1[SORCS2_1$Sex == 0,][, "SEX"] <- 1   #M=0
SORCS2_1[SORCS2_1$Sex == 1,][, "SEX"] <- 2   #F=1

SORCS2_1[SORCS2_1$APOE == 22,][, "APOE4dose"] <- 0
SORCS2_1[SORCS2_1$APOE == 23,][, "APOE4dose"] <- 0
SORCS2_1[SORCS2_1$APOE == 24,][, "APOE4dose"] <- 1
SORCS2_1[SORCS2_1$APOE == 33,][, "APOE4dose"] <- 0
SORCS2_1[SORCS2_1$APOE == 34,][, "APOE4dose"] <- 1
SORCS2_1[SORCS2_1$APOE == 44,][, "APOE4dose"] <- 2

SORCS2_1[SORCS2_1$APOE == 22,][, "APOE2dose"] <- 2
SORCS2_1[SORCS2_1$APOE == 23,][, "APOE2dose"] <- 1
SORCS2_1[SORCS2_1$APOE == 24,][, "APOE2dose"] <- 1
SORCS2_1[SORCS2_1$APOE == 33,][, "APOE2dose"] <- 0
SORCS2_1[SORCS2_1$APOE == 34,][, "APOE2dose"] <- 0
SORCS2_1[SORCS2_1$APOE == 44,][, "APOE2dose"] <- 0

SORCS2_1[SORCS2_1$Series == "ADSP",][, "ADSP"] <- 1
#SORCS2[SORCS2$Series == "SibPair_White",][, "JS"] <- 1
#SORCS2[SORCS2$Series == "RS",][, "RS"] <- 1
#SORCS2[SORCS2$Series == "AUT",][, "AUT"] <- 1
#SORCS2[SORCS2$Series == "Indiana",][, "NCRAD"] <- 1
#SORCS2[SORCS2$Series == "NW",][, "NW"] <- 1

#y <- SORCS2 %>% filter(rs=="rs6816604",JS==1) %>% group_by(rs,Dx,A1,A2) %>% summarize(n=n())
#print(y)

        #Create PLINK fam file
fam <- select(SORCS2_1,FID,IID,PID,MID,SEX,AFF)
fam <- distinct(fam)
str(fam); head(fam); tail(fam)
write.table(fam,out2,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2fam_1=read.table("SORCS2_1.fam", header = F)
#str(SORCS2fam_1);head(SORCS2fam_1);tail(SORCS2fam_1)

        #Create PLINK cov file; 
        #Not sure why, but errors sometimes occurred in r when using "SORCS2.cov"" as output using write.table
        #So I used "SORCS2cov.txt" thereby avoiding this problem
cov <- select(SORCS2_1,FID,IID,ADSP,SEX,Age,APOE4dose,APOE2dose)
cov <- distinct(cov)
str(cov)
write.table(cov,out3,sep='\t',quote=FALSE,row.names=F,col.names=T)
#SORCS2cov_1=read.table("SORCS2_1cov.txt", header = T)
#str(SORCS2cov_1);head(SORCS2cov_1); tail(SORCS2cov_1)

        #Create PLINK lgen file
lgen <- select(SORCS2_1,FID,IID,SNP_ID,A1,A2)
str(lgen)
write.table(lgen,out4,sep='\t',quote=FALSE,row.names=F,col.names=F)
#SORCS2lgen_1=read.table("SORCS2_1.lgen", header = F)
#str(SORCS2lgen_1);head(SORCS2lgen_1);tail(SORCS2lgen_1)
}
#ADSP2plink(,out1="ADSP.map", out2="ADSP.fam", out3="ADSPcov.txt", out4="ADSP.lgen")
list.files()


        #PLINK --assoc: analyzes standard case/control (allelic) association with LOAD, no covariates
#system("plink --lfile ADSP --assoc  --out ADSPassoc")
#ADSPassoc <- read.table("ADSPassoc.assoc", header = T)
#str(ADSPassoc);head(ADSPassoc);tail(ADSPassoc)

        #PLINK --assoc: analyzes association by logistic regression using an additive model with with covariates
#system("plink --lfile ADSP --logistic --ci 0.95 --covar ADSPcov.txt --covar-name SEX,APOE4dose,APOE2dose --out ADSPlogAllCov")
#y <- read.table("ADSPlogAllCov.assoc.logistic", header=T)
#print.data.frame(y)
#print.data.frame(x)
#a<- read.table("ADSPcov.txt", header=T)
#print.data.frame(a)
```
print.data.frame(x)
print.data.frame(y)
```
#system("plink --lfile SORCS2 --hap-window 4 --hap-logistic --ci 0.95 --covar SORCS2cov.txt --covar-name RS-APOE2dose --mhf 0.005 --out SORCS2haplogAllCov")
#system("plink --lfile SORCS2 --hap-window 4 --hap-logistic --ci 0.95 --covar SORCS2cov.txt --covar-name RS-APOE2dose --mhf 0.005 --hap-omnibus --out #SORCS2haplogAllCovOmni")
system("plink --lfile SORCS2 --logistic --ci 0.95 --covar SORCS2cov.txt --covar-name APOE4dose,APOE2dose --out SORCS2logAPOECov")
list.files()







###Ben and Leonie's Content
First draft of gene diagram - exact positions and lengths of exons to be added.
```{r}
# set up the plot
op <- par(bg = "white")
plot(c(0, 550230), c(-100, 450), main="Gene Diagram \n exons and respective domains", type= "n", xlab="position on gene", ylab="",yaxt='n',frame.plot=FALSE)
rect(0, 25, 550230, 75, col="grey", border="grey") 

#exons and respective domains
##signal sequence
rect(10000, 0, 11000, 100, col="green", border="green3") 
rect(39000, 0, 40000, 100, col="green", border="green3")
text(30000,150, labels= "Signal sequence",cex=0.5,col="green3")

##propeptide
rect(80000, 0, 81000, 100, col="cyan", border="cyan3")
rect(109000, 0, 110000, 100, col="cyan", border="cyan3")
text(90000,180, labels= "Propeptide",cex=0.5,col="cyan3")

##vps10p domain
rect(150000, 0, 151000, 100, col="blue", border="blue4") 
rect(160000, 0, 161000, 100, col="blue", border="blue4") 
rect(189000, 0, 190000, 100, col="blue", border="blue4") 
rect(199000, 0, 200000, 100, col="blue", border="blue4") 
rect(209000, 0, 210000, 100, col="blue", border="blue4") 
rect(219000, 0, 220000, 100, col="blue", border="blue4") 
text(190000,150, labels= "VPS10P domain",cex=0.5,col="blue4")

##pkd domain
rect(280000, 0, 281000, 100, col="red", border="red3") 
rect(290000, 0, 291000, 100, col="red", border="red3") 
rect(309000, 0, 310000, 100, col="red", border="red3")
rect(319000, 0, 320000, 100, col="red", border="red3")
text(300000,180, labels= "PKD domain",cex=0.5,col="red3")

##lrr
rect(370000, 0, 371000, 100, col="magenta", border="magenta3")
rect(380000, 0, 381000, 100, col="magenta", border="magenta3")
rect(389000, 0, 390000, 100, col="magenta", border="magenta3") 
rect(399000, 0, 400000, 100, col="magenta", border="magenta3") 
text(380000,150, labels= "LRR",cex=0.5,col="magenta3")

##tm
rect(450000, 0, 451000, 100, col="yellow", border="yellow3") 
rect(469000, 0, 470000, 100, col="yellow", border="yellow3") 
rect(479000, 0, 480000, 100, col="yellow", border="yellow3") 
text(470000,180, labels= "TM",cex=0.5,col="yellow3")

##c-tail
rect(510000, 0, 511000, 100, col="orange", border="orange3")
rect(519000, 0, 520000, 100, col="orange", border="orange3") 
rect(539000, 0, 540000, 100, col="orange", border="orange3")
text(520000,180, labels= "C-tail",cex=0.5,col="orange3")

```

###Joseph's Content
ADSP data was processed using the Genome GPS pipeline by the Bioinformatics Core in Rochestor. After variant calling (using 'GenotypeGVCFs' in the GATK pipeline), variants were annotated using BioR.  
Variants from genes of interest (here, SORCS2) were identified and processed using Perl scripts to generate files for downstream analysis.

```{r}
system("perl generate_flat_file_SG.pl ADSP ADSP_sample_sex_age_AD_APOE.txt SORCS2_variants.txt 1000 SORCS2_ff.txt SORCS2_allele_counts.txt")

#Sex : 0: Male, 1: Female
#Case/Control: 0: Control, 1: AD, 2: Probable AD
#Call: 11/12/22
#Genotype: 0/0, 0/1, 1/1,  [./., ./0, ./1, 0/., 1/. encoded as NA]

#Importing Flat File
SORCS2_FF <- read.table("SORCS2_ff.txt",header = TRUE, check.names = FALSE)
head(SORCS2_FF)

#Importing allele counts for each variant
SORCS2_variants <- read.table("SORCS2_allele_counts.txt", header = TRUE, check.names = FALSE)
head(SORCS2_variants)
```
y


